File: src/main/java/com/backend/immilog/image/infrastructure/persistence/jpa/entity/ImageEntity.java
package com.backend.immilog.image.infrastructure.persistence.jpa.entity;

import com.backend.immilog.image.domain.enums.ImageStatus;
import com.backend.immilog.image.domain.enums.ImageType;
import com.backend.immilog.image.domain.model.Image;
import jakarta.persistence.*;
import org.hibernate.annotations.DynamicUpdate;

@DynamicUpdate
@Entity
@Table(name = "image")
public class ImageEntity {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "seq")
    private Long seq;

    @Column(name = "path")
    private String path;

    @Column(name = "image_type")
    @Enumerated(EnumType.STRING)
    private ImageType imageType;

    @Column(name = "status")
    @Enumerated(EnumType.STRING)
    private ImageStatus status;

    protected ImageEntity() {}

    protected ImageEntity(
            Long seq,
            String path,
            ImageType imageType,
            ImageStatus status
    ) {
        this.seq = seq;
        this.path = path;
        this.imageType = imageType;
        this.status = status;
    }

    public static ImageEntity from(Image image) {
        return new ImageEntity(
                image.seq(),
                image.path(),
                image.imageType(),
                image.status()
        );
    }

    public Image toDomain() {
        return new Image(
                this.seq,
                this.path,
                this.imageType,
                this.status
        );
    }
}

File: src/main/java/com/backend/immilog/image/infrastructure/persistence/jpa/entity/ImageEntity.kt
package com.backend.immilog.image.infrastructure.persistence.jpa.entity

import com.backend.immilog.image.domain.enums.ImageStatus
import com.backend.immilog.image.domain.enums.ImageType
import com.backend.immilog.image.domain.model.Image
import jakarta.persistence.*
import org.hibernate.annotations.DynamicUpdate

@DynamicUpdate
@Entity
@Table(name = "image")
open class ImageEntity(
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "seq")
    open var seq: Long? = null,

    @Column(name = "path")
    open var path: String = "",

    @Column(name = "image_type")
    @Enumerated(EnumType.STRING)
    open var imageType: ImageType = ImageType.POST,

    @Column(name = "status")
    @Enumerated(EnumType.STRING)
    open var status: ImageStatus = ImageStatus.NORMAL
) {
    companion object {
        fun from(image: Image): ImageEntity =
            ImageEntity(
                seq = image.seq,
                path = image.path,
                imageType = image.imageType,
                status = image.status
            )
    }

    fun toDomain(): Image = Image(
        seq = this.seq,
        path = this.path,
        imageType = this.imageType,
        status = this.status
    )
}
